services/agent/tests/unit/test_info.py

```diff
@@ -0,0 +1,36 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+import asyncio
+from contextlib import suppress
+from typing import AsyncIterator, Final
+
+import pytest
+from pydantic import PositiveFloat
+from simcore_service_agent._app import Application
+from simcore_service_agent.info._consumer import _async_request_info
+from simcore_service_agent.info._provider import info_exposer
+
+WAIF_FOR_SERVER_TO_START: Final[PositiveFloat] = 0.1
+
+
+@pytest.fixture
+async def provider() -> AsyncIterator[None]:
+    app = Application()
+    task = asyncio.create_task(info_exposer(app))
+
+    await asyncio.sleep(WAIF_FOR_SERVER_TO_START)
+
+    yield
+
+    task.cancel()
+    with suppress(asyncio.CancelledError):
+        await task
+
+
+async def test_ok(provider: None):
+    result = await _async_request_info()
```

Comment:
Suggestion: to echo from @pcrespov comment on the healthcheck function, you could use [https://pypi.org/project/pytest-benchmark/](https://pypi.org/project/pytest-benchmark/) (we already use it), to benchmark the healthcheck and see speed, and maybe also CPU consumption.
Does not give out CPU consumption sadly. but maybe there is another pytest package that does it. that could be handy for other things as well..

Reply:
maybe that one [https://pypi.org/project/pytest-monitor/](https://pypi.org/project/pytest-monitor/)?

Reply:
this was removed in favour of fastapi. Also I'd use something like https://github.com/plasma-umass/scalene